pkg/receive/multitsdb.go

```diff
@@ -127,9 +134,13 @@ func NewMultiTSDB(
 		bucket:                bucket,
 		allowOutOfOrderUpload: allowOutOfOrderUpload,
 		hashFunc:              hashFunc,
+		matcherCache:          storecache.NewNoopMatcherCache(),
 	}
 
 	for _, option := range options {
+		if option == nil {
```

Comment:
When can this be nil?

Reply:
there were a bunch of tests calling this with nil. fixed them now.